test/common/router/config_impl_test.cc

```diff
@@ -2008,6 +2008,91 @@ TEST_F(RouteMatcherTest, HeaderMatchedRouting) {
   }
 }
 
+TEST_F(RouteMatcherTest, UrlTemplateMatchedRouting) {
+  const std::string yaml = R"EOF(
+virtual_hosts:
+- name: local_service
+  domains:
+  - "*"
+  routes:
+  - match:
+      url_template: "/foo/bar"
+      headers:
+      - name: test_header
+        exact_match: test
+    route:
+      cluster: foo_bar_with_header
+  - match:
+      url_template: "/foo/bar"
+    route:
+      cluster: foo_bar
+  - match:
+      url_template: "/foo/*"
+      headers:
+      - name: test_header
+        exact_match: test
+    route:
+      cluster: foo_1_with_header
+  - match:
+      url_template: "/foo/*"
+    route:
+      cluster: foo_1
+  - match:
+      url_template: "/foo/**"
+      headers:
+      - name: test_header
+        exact_match: test
+    route:
+      cluster: foo_11_with_header
+  - match:
+      url_template: "/foo/**"
+    route:
+      cluster: foo_11
+  )EOF";
```

Comment:
I guess a never reached template "/foo/baz" could be appended here for illustrative purposes.

Reply:
Done